### PR TITLE
Add KafkaRest as new output for Falco Sidekick

### LINKED ISSUE #262: Support Kafka Rest Output
**Motivation**

The Kafka REST Proxy provides a RESTful interface to a Apache Kafka cluster, making it easy to produce and consume messages, view the state of the cluster, and perform administrative actions without using the native Kafka protocol or clients.

**Feature**

Having a new Output of type `kafkarest` in addition to Kafka.

```yaml
kafkarest:
  address: "" # The full URL to the topic (example "http://kafkarest:8082/topics/test")
  #version: 2 # Kafka Rest Proxy API version 2|1 (default: 2)
  # minimumpriority: "debug" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
  # mutualtls: false # if true, checkcert flag will be ignored (server cert will always be checked)
  # checkcert: true # check if ssl certificate of the output is valid (default: true)
```

The `Content-Type: application/vnd.kafka.binary.v2+json` will produce the key value pairs exactly as passed them, without adding escaped quotes to string keys.

On note, why a normal Webhook is not suitable neither, Kafka Rest Proxy expects the data on following message body:
```json
{"records":[{"value":"falcoPayload"}]}
```
**Alternatives**

Using the existing Kafka Output, but need to extend it to support cert or sasl. This is realy not the easy. I would rather let it like this and use the rest proxy as an entry. The Rest proxy user can be set specific via ACL on the topic.

**Additional context**

Here the Kafka Rest Output in action
image

And the result in Aiven UI
image

### PR DESCRIPTION
**Pull request by dirien**
Signed-off-by: Engin Diri <engin.diri@mail.schwarz>

**What type of PR is this?**

Fixes #262 

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area config

/area outputs

> /area tests

**What this PR does / why we need it**:
The Kafka REST Proxy provides a RESTful interface to a Apache Kafka cluster, making it easy to produce and consume messages, view the state of the cluster, and perform administrative actions without using the native Kafka protocol or clients.

**Which issue(s) this PR fixes**:  #262 

Fixes #262 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Issif** [comment]
Thank you very much, this is a very valuable PR, with a really nice description. I'll review asap but a first look tells me it's already really good.

👍

File: config.go [hunk 1]
```diff
@@ -352,6 +358,7 @@ func getConfig() *types.Configuration {
 	c.GCP.CloudRun.MinimumPriority = checkPriority(c.GCP.CloudRun.MinimumPriority)
 	c.Googlechat.MinimumPriority = checkPriority(c.Googlechat.MinimumPriority)
 	c.Kafka.MinimumPriority = checkPriority(c.Kafka.MinimumPriority)
+	c.KafkaRest.MinimumPriority = checkPriority(c.Kafka.MinimumPriority)
```
Comment (Issif) [inline config.go:361]:
```suggestion
	c.KafkaRest.MinimumPriority = checkPriority(c.KafkaRest.MinimumPriority)
```

File: main.go [hunk 2]
```diff
@@ -361,6 +362,16 @@ func init() {
 		}
 	}
 
+	if config.KafkaRest.Address != "" {
+		var err error
+		kafkaRestClient, err = outputs.NewClient("KafkaRest", config.KafkaRest.Address, config.KafkaRest.MutualTLS, config.KafkaRest.CheckCert, config, stats, promStats, statsdClient, dogstatsdClient)
+		if err != nil {
+			config.KafkaRest.Address = ""
+		} else {
+			outputs.EnabledOutputs = append(outputs.EnabledOutputs, "Kafka Rest")
```
Comment (Issif) [inline main.go:371]:
```suggestion
			outputs.EnabledOutputs = append(outputs.EnabledOutputs, "KafkaRest")
```
outputs are separated by a space when we print the list, this is why we need a string in camel case

File: outputs/kafkarest.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	case 2:
+		version = c.Config.KafkaRest.Version
+	case 1:
+		version = c.Config.KafkaRest.Version
+	default:
+		version = 2
+	}
+	falcoMsg, err := json.Marshal(falcopayload)
+	if err != nil {
+		c.setKafkaErrorMetrics()
```
Comment (Issif) [inline outputs/kafkarest.go:37]:
```suggestion
	    c.Stats.KafkaRest.Add(Error, 1)
		c.PromStats.Outputs.With(map[string]string{"destination": "kafkarest", "status": Error}).Inc()
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Comment by dirien** [comment]
@Issif, Thanks for the feedback. I changed everything according to your suggestions. looking forward. 😄

**Review by Issif — CHANGES_REQUESTED** [review]
Only 1 thing is missing, can you add your output in this [list](https://github.com/falcosecurity/falcosidekick#message-queue--streaming) please

**Comment by dirien** [comment]
@Issif, done :) during the  Yandex rebase it must slipped through 😓

**Review by Issif — APPROVED** [review]
Congrats and thank you !!

**Pull request merged**